### PR TITLE
Update model specs which bases vm creation.

### PR DESCRIPTION
**Pull request by zongzw**

### CODE REVIEW

File: app/waf/src/models/adc.model.ts [hunk 1]
```diff
@@ -1,69 +1,180 @@
...
+    type: 'object',
+    required: true,
     schema: {
-      create: true,
       response: true,
-      example: 'admin',
     },
   })
-  username: string;
+  management: ConfigTypes['management'];
```
Comment (zhaoqin-github) [inline app/waf/src/models/adc.model.ts]: Is it the management IP/port generated by you for user to read?

File: app/waf/src/models/adc.model.ts [hunk 2]
```diff
@@ -1,69 +1,180 @@
...
+  //   required: true,
+  // })
+  // onBoarding: ConfigBody['onBoarding'];
 
   constructor(data?: Partial<Adc>) {
     super(data);
   }
 }
+
+export class ActionsResponse extends Entity {
```
Comment (zhaoqin-github) [inline app/waf/src/models/adc.model.ts]: Not quite sure what will its response looks like now. Usually most action APIs may return 204, and its effect will be displayed by resource content.

File: app/waf/src/models/adc.model.ts [hunk 3]
```diff
@@ -1,69 +1,180 @@
-import {model, property} from '@loopback/repository';
+import {model, property, Entity} from '@loopback/repository';
 import {CommonEntity} from '.';
 
+export type ActionsBody = CreateBody & DeleteBody & SetupBody;
+
+type CreateBody = {
+  create: {
+    id: string;
```
[the remark this answers is not in the document]
Comment (zongzw) [reply app/waf/src/models/adc.model.ts]: adc id.

File: app/waf/src/models/adc.model.ts (hunk 1 above)
Comment (zongzw) [reply app/waf/src/models/adc.model.ts]: The IP / Port can be either generated by us or appointed by user. 

The IP is actually 'mgmt' type fixed ip from networks.

File: app/waf/src/models/adc.model.ts (hunk 2 above)
Comment (zongzw) [reply app/waf/src/models/adc.model.ts]: Just removed it and put this code in another pr which contains API spec. You can decide there how to define the schema.

File: app/waf/src/models/adc.model.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (zhaoqin-github) [reply app/waf/src/models/adc.model.ts]: The adc id is located in url path. Why do we need to place it in action request body?

[the remark this answers is not in the document]
Comment (zongzw) [reply app/waf/src/models/adc.model.ts]: Not true for our API spec design. Please review that slides again.

[the remark this answers is not in the document]
Comment (zhaoqin-github) [reply app/waf/src/models/adc.model.ts]: I will propose to change that action API to 'POST /adcaas/v1/adcs/{adcId}/actions'. That is the usually way to make a rest call.

[the remark this answers is not in the document]
Comment (zongzw) [reply app/waf/src/models/adc.model.ts]: Agree. Will do it in the coming PR which refactors AdcController.

**Pull request merged**